### PR TITLE
India - Periodic Labour Force Survey (PLFS) - Average wage/salary earnings

### PR DESCRIPTION
**Pull request by thejeshgn**
- Added 2018-2019 data
- Added 2017-2018 data

---
This change is [Reviewable](https://reviewable.io/reviews/datacommonsorg/data/386)

### CODE REVIEW

File: scripts/india_plfs/wage_data/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Periodic Labour Force Survey (PLFS) - Average wage/salary earnings
+
+
+## About the Dataset
+Average wage/salary earnings (Rs. 0.00) during the preceding calendar month from regular wage/salaried employment among the regular wage/ salaried employees in CWS for each State/ UT.
+
+### Download URL
+
+* PLES July 2018 - June 2019 [XLSX](http://mospi.nic.in/sites/default/files/reports_and_publication/PLFS_2018_19_Anual/Table_42.xlsx)
```
Comment (pradh) [inline scripts/india_plfs/wage_data/README.md:9]: How often is this data released?  Is there data older than this?

(Generally, if the format stays mostly same, its good to get data from as many years as there exists.)

File: scripts/india_plfs/wage_data/preprocess_test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+            loader = PLFSWageDataLoader(xlsx_file, period="2018-10")
+            loader.load()
+            loader.process()
+            loader.save(csv_file_path=result_file)
+
+            same = filecmp.cmp(result_file, expected_file)
+            os.remove(result_file)
+
+        self.assertTrue(same)
```
Comment (pradh) [inline scripts/india_plfs/wage_data/preprocess_test.py:45]: I know we use this pattern elsewhere too, but its harder to tell the diff when it fails.

Consider reading into memory the two files, and doing:
  self.assertEqual(..., ...)

(But feel free to make the change in a separate PR)

**Review by pradh — APPROVED** [review]

**Review by pradh — COMMENTED** [review]
This is looking good, I used the following link to verify the import:

https://datacommonsorg.github.io/tools/mcf-browser/#&file=https://raw.githubusercontent.com/datacommonsorg/data/ef8efc596cae0885f848c7563f30464f2805477c/scripts/india_plfs/wage_data/PLFSWageData_India.tmcf&file=https://raw.githubusercontent.com/datacommonsorg/data/ef8efc596cae0885f848c7563f30464f2805477c/scripts/india_plfs/wage_data/PLFSWageData_India_Table_42_10_12_2018.csv&file=https://raw.githubusercontent.com/datacommonsorg/data/ef8efc596cae0885f848c7563f30464f2805477c/scripts/india_plfs/wage_data/PLFSWageData_India_StatisticalVariables.mcf

File: scripts/india_plfs/wage_data/README.md (hunk 1 above)
Comment (thejeshgn) [reply scripts/india_plfs/wage_data/README.md]: Usually once a year. I am trying to find older reports to extract the data. The format seems to be same.
 I added 2017-2018. I will add as I find more.

**Review by pradh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/data/386)* status: 0 of 26 files reviewed, 2 unresolved discussions (waiting on @thejeshgn)

___
*[scripts/india_plfs/wage_data/README.md, line 9 at r1](https://reviewable.io/reviews/datacommonsorg/data/386#-MSTMzcj7Dg12f-d3xUj-r1-9:-MSTNKJV-SEUgdqJp6DH:bvs7lve) ([raw file](https://github.com/datacommonsorg/data/blob/ef8efc596cae0885f848c7563f30464f2805477c/scripts/india_plfs/wage_data/README.md#L9)):*
<details><summary><i>Previously, thejeshgn (Thejesh GN) wrote…</i></summary><blockquote>

Usually once a year. I am trying to find older reports to extract the data. The format seems to be same.
 I added 2017-2018. I will add as I find more. 
</blockquote></details>

SG. Thanks.

File: scripts/india_plfs/wage_data/preprocess.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            "data/{data_file}.xlsx".format(data_file=data_file),
+        )
+        loader = PLFSWageDataLoader(data_file_path, period)
+        csv_file_path = os.path.join(
+            os.path.dirname(__file__),
+            "./PLFSWageData_India_{data_file}.csv".format(data_file=data_file))
+
+        loader.load()
+        loader.process()
+        loader.save(csv_file_path)
```
Comment (pradh) [inline scripts/india_plfs/wage_data/preprocess.py:199]: Thinking about importing the, could we change it to output a single CSV file?

The downstream pipeline configs will be a lot simpler.

Thanks!

Comment (thejeshgn) [reply scripts/india_plfs/wage_data/preprocess.py:199]: Done.

https://github.com/datacommonsorg/data/pull/390

**Pull request merged**